projects/plugins/jetpack/extensions/blocks/blogroll/components/blogroll-appender/index.js

```diff
@@ -14,11 +15,29 @@ export default function BlogrollAppender( { subscriptions, clientId } ) {
 	const [ popoverAnchor, setPopoverAnchor ] = useState();
 	const [ searchInput, setSearchInput ] = useState( '' );
 	const { insertBlock } = dispatch( 'core/block-editor' );
+	const { getBlocks } = useSelect( blockEditorStore );
+
+	const blogrollBlock = getBlocks( clientId );
 	const { siteDetails } = useGetSiteDetails( {
 		siteURL: searchInput,
 		subscriptions,
 		enabled: searchInput,
 	} );
+
+	// Check if site is already appended to the blogroll
+	// If it is, add a duplicateRecommendation flag to the site object
+	const blogrollItems = blogrollBlock?.filter( block => block.name === 'jetpack/blogroll-item' );
+	const updatedBlogrollItems = siteDetails?.map( site => {
+		const itemBlogId = site?.blog_id;
+		const itemBlogrollItem = blogrollItems.find( item => {
+			return item?.attributes.id === itemBlogId;
+		} );
+		if ( itemBlogrollItem ) {
+			return { ...site, duplicateRecommendation: true };
+		}
+		return site;
+	} );
+
```

Comment:
A cleaner way to get this info is like so:

```js
const [ siteRecommendations ] = useEntityProp(
	'root',
	'site',
	'Blogroll Recommendations'
);
```

We store this info on [every change](https://github.com/Automattic/jetpack/blob/8eef2eb53e93310e95eea1fd369048d85d72cf98/projects/plugins/jetpack/extensions/blocks/blogroll/use-site-recommendations.js#L10-L14).

And I would pass the items as-is and determine duplication inside the `BlogrollAppenderResults`. The Appender doesn't need to know about this logic.